os/arch/arm/src/amebad/amebad_serial.c

```diff
@@ -423,36 +425,24 @@ static void rtl8721d_up_shutdown(struct uart_dev_s *dev)
  *
  ****************************************************************************/
 
-//extern uint32_t uart_irqhandler(void *data);
 void rtl8721d_uart_irq(uint32_t id, SerialIrq event)
 {
 	struct uart_dev_s *dev = (struct uart_dev_s *)id;
 	struct rtl8721d_up_dev_s *priv = (struct rtl8721d_up_dev_s *)dev->priv;
-	if(event == RxIrq) {
+	if (event == RxIrq) {
 		uart_recvchars(dev);
 	}
-	if(event == TxIrq){
+	if (event == TxIrq) {
 		uart_xmitchars(dev);
 	}
 }
+
 static int rtl8721d_up_attach(struct uart_dev_s *dev)
 {
 	struct rtl8721d_up_dev_s *priv = (struct rtl8721d_up_dev_s *)dev->priv;
 	int ret = 0;
 	DEBUGASSERT(priv);
-#if 0
-	/* Attach and enable the IRQ */
-	ret = irq_attach(priv->irq, rtl8721d_up_interrupt, NULL);		//UART_LOG_IRQ(19)
-	if (ret == OK) {
-		/* Enable the interrupt (RX and TX interrupts are still disabled
-		 * in the UART
-		 */
-
-		up_enable_irq(priv->irq);
-	}
-#else
-	serial_irq_handler(sdrv[uart_index_get(priv->Tx)], rtl8721d_uart_irq, (uint32_t)dev);
-#endif
+	serial_irq_handler(sdrv[uart_index_get(priv->tx)], rtl8721d_uart_irq, (uint32_t) dev);
```

Comment:
Is the UART interrupt working with these changes ??

Reply:
HAve updated only UART test case changes in new commit